addons/web/static/src/views/fields/copy_clipboard/copy_button.js

```diff
@@ -1,33 +1,46 @@
 /** @odoo-module **/
 
-const { Component, useEffect, useState } = owl;
+import { browser } from "@web/core/browser/browser";
+import { Tooltip } from "@web/core/tooltip/tooltip";
+import { useService } from "@web/core/utils/hooks";
 
+const { Component, onMounted, useRef } = owl;
 export class CopyButton extends Component {
     setup() {
-        this.state = useState({
-            isCopied: false,
+        this.button = useRef("button");
+        this.popover = useService("popover");
+        this.writePermission = false;
+        onMounted(async () => {
+            this.writePermission = await browser.navigator.permissions.query({ name: 'clipboard-write' });
         });
-        useEffect(
-            () => {
-                return () => clearTimeout(this.timeoutId);
-            },
-            () => [this.state.isCopied]
+    }
+
+    showTooltip() {
+        const closeTooltip = this.popover.add(
+            this.button.el,
+            Tooltip,
+            { tooltip: this.props.successText },
         );
+        browser.setTimeout(() => {
+            closeTooltip();
+        }, 800)
     }
 
     async onClick() {
+        if (!this.writePermission) {
+            return console.info("Browser doesn't allow to copy to clipboard");
+        }
```

Comment:
the return value for writePermission is a string which is one of "granted", "denied" or "prompt", you have to handle it accordingly if you're going to handle it at all.

Reply:
Oops... my bad. It should now be correct